### ISSUE TITLE
Verify if tokenizer is saved during training and ready for Hugging Face upload

### ISSUE THREAD

**Issue by chrisjcc** [chrisjcc/text-to-sql-finetuning#24]
**Assigned to:** @claude
**Priority:** High
**Labels:** investigation, training, huggingface, tokenizer

**Context:**
We’ve confirmed that the trained adapter model is being saved and later uploaded to Hugging Face using scripts/upload_to_hf.py. However, it’s unclear whether the tokenizer used during training is being saved as part of the process (e.g. via

```
tokenizer.save_pretrained(output_dir)
```

If the tokenizer isn’t saved, the uploaded model on Hugging Face will be incomplete and unusable for inference or further fine-tuning, since the correct tokenization mapping won’t be available. Note: Two special tokens were added.

---

#### **Tasks**
1. Inspect the training pipeline (e.g. trainer, model saving step) and verify:
   - Where the model artifacts are saved.
   - Whether the tokenizer is explicitly saved via `tokenizer.save_pretrained(...)`.
2. If not saved, implement a step to persist it alongside the adapter weights.
3. Confirm that `scripts/upload_to_hf.py` also uploads the tokenizer directory (usually under the same `output_dir` as the model).
4. Document the verified behavior in the README or `training.md`.

---

#### **Acceptance Criteria**
- [ ] `tokenizer.json`, `vocab.json`, `merges.txt`, or `tokenizer_config.json` are present in the output directory.
- [ ] Hugging Face upload script includes the tokenizer files.
- [ ] Model card or documentation notes which tokenizer version was used.

---

**Notes**
Tokenizer consistency between training and inference is critical. Without saving and uploading the tokenizer, anyone pulling the model from Hugging Face will not be able to reproduce or run inference correctly.

---
